frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx

```diff
@@ -146,24 +146,28 @@ export default function CreateCluster(props: { infrastructureType: ClusterInfras
   const [selectedConnection, setSelectedConnection] = useState<ProviderConnection>()
   const onControlChange = useCallback(
     (control: any) => {
-      if (control.id === 'connection') {
-        if (newSecret && control.setActive) {
-          const secretName = newSecret?.metadata.name ?? ''
-          if (control.providerId === 'kubevirt') {
-            // preset replacement fields to get around delayed control state from setAvailableConnections
-            control.availableMap[secretName] = {
-              replacements: {
-                pullSecret: newSecret.data?.pullSecret ?? '',
-                'ssh-publickey': newSecret.data?.['ssh-publickey'] ?? '',
-                encoded: true,
-              },
-            }
+      if (newSecret && control.setActive) {
+        const secretName = newSecret?.metadata.name ?? ''
+
+        // Handle new secret when it's available
+        if (control.providerId === 'kubevirt') {
+          // preset replacement fields to get around delayed control state from setAvailableConnections
+          control.availableMap[secretName] = {
+            replacements: {
+              pullSecret: newSecret.data?.pullSecret ?? '',
+              'ssh-publickey': newSecret.data?.['ssh-publickey'] ?? '',
+              externalInfraKubeconfig: newSecret.data?.externalInfraKubeconfig ?? '',
+              externalInfraNamespace: newSecret.data?.externalInfraNamespace ?? '',
+              encoded: true,
+            },
           }
-          control.setActive(secretName)
-          setNewSecret(undefined) // override with the new secret once
         }
+        control.setActive(secretName)
+        setNewSecret(undefined) // Reset the new secret once it's used
+      }
+      if (control.id === 'connection') {
```

Comment:
I think I need help to understand why you moved code outside of the `if (control.id === 'connection') {` block. This `onControlChange` will run any time any control is changed. If `newSecret` is set, I think it is nearly guaranteed that the control that just changed is the connection, but I'm not sure why we would risk it.

Reply:
you are right @KevinFCormier. I refactored something in here a couple of days back and moved it out of the `if` block. thanks for catching this. I will move it back to within the outer `if block